### ISSUE TITLE
[install] Install of development files should improve.

### ISSUE THREAD

**Issue by ejgallego** [coq/coq#6291]
In current `v8.7/master`, `make install` won't yield a complete development environment for plugins due to 2 issues:

- cmx files are not installed in `coqlib`:
  This yields the warning:
  `Warning 58: no cmx file was found in path for module Sorts, and its interface was not compiled with -opaque`
  which is pretty annoying.

  I guess the obvious solution is to install such files. See also #5099 

- In addition to the missing `cmx` files, we are not installing the needed `.o` files to support linking the plugins statically. IMHO, such files should be provided as linking a plugin statically is a 100% legitimate use case.

IMHO these bugs are critical enough as not to release 8.7.1 without a fix, as otherwise using the shipped `META` file for some advanced cases is hard.

**Comment by Zimmi48** [comment]
I'd like 8.7.1 not to be blocked by a blocker bug that nobody is working on.
Anyone willing to fix this?

**Comment by ejgallego** [comment]
I am working on this.

**Comment by ejgallego** [comment]
I'd appreciate hearing opinions thou on what should the proper fix be.

**Comment by ejgallego** [comment]
The pull request fixes the two first items, I think I will split the third item out to a new bug as it is not critical.

**Comment by Zimmi48** [comment]
@ejgallego I am not closing this right now to remind you that you want to split the third item into a new issue.

**Comment by ejgallego** [comment]
Let's clos e this, the third item is in my todo.
